testsuite/gateway/gateway_api/route.py

```diff
@@ -114,3 +114,15 @@ def add_backend(self, backend: "Backend", prefix="/"):
     @modify
     def remove_all_backend(self):
         self.model.spec.rules.clear()
+
+    def wait_for_ready(self):
+        """Waits until HTTPRoute is reconcilled by GatewayProvider"""
+
+        def _ready(obj):
+            for condition_set in obj.model.status.parents:
+                if condition_set.controllerName == "istio.io/gateway-controller":
+                    return (all(x.status == "True" for x in condition_set.conditions),)
```

Comment:
`all` returns `True` in case of empty collection (`condition_set.conditions` in this case). Is it ok in this context or better to add check that `len(condition_set.conditions) > 0`

Reply:
I think it should be fine, only potential issue is if the condition set was added but has no conditions when we check. I find it unlikely (I sincerely hope they are adding the condition set with condition at once, otherwise it would not make much sense tome) but I can add it if you want

Reply:
I think you're right that it is quite unlikely, no need to add.